python/pyarrow/filesystem.py

```diff
@@ -87,20 +87,10 @@ def read_parquet(self, path, columns=None, metadata=None, schema=None,
         -------
         table : pyarrow.Table
         """
-        from pyarrow.parquet import read_multiple_files
-
-        if self.isdir(path):
-            paths_to_read = []
-            for path in self.ls(path):
-                if path.endswith('parq') or path.endswith('parquet'):
-                    paths_to_read.append(path)
-        else:
-            paths_to_read = [path]
-
-        return read_multiple_files(paths_to_read, columns=columns,
-                                   filesystem=self, schema=schema,
-                                   metadata=metadata,
-                                   nthreads=nthreads)
+        from pyarrow.parquet import ParquetDataset
```

Comment:
having ``read_parquet`` in ``FileSystem`` which is almost an ABC is a bit odd, though ATM not sure of a better way to organize.